ereolen_support/ereolen_support.helper.inc

```diff
@@ -14,34 +14,39 @@
  * @return string|null
  *   Account id of the user.
  */
-function _ereolen_search_user(string $email): ?string {
-  $accountId = NULL;
-  $requestParts = _ereolen_support_get_request_parts('/user/search?query=' . $email, 'GET');
-  $searchQueryResult = _ereolen_support_callApi($requestParts);
-  // Endpoint searches in all fields, so we must ensure email match.
+function _ereolen_support_search_user(string $email): ?string {
+  $searchQueryResult = _ereolen_support_call_api('/user/search?query=' . $email, 'GET');
+  // Endpoint searches in all fields, so we must ensure email matches.
   foreach ($searchQueryResult['content'] as $match) {
+    // Atlassian accounts have their email stripped from user results,
+    // so in that case we have no email to match up against.
     if ($match->emailAddress === $email || (empty($match->emailAddress) && $match->accountType === 'atlassian')) {
-      $accountId = $match->accountId;
+      return $match->accountId;
     }
   }
 
-  return $accountId;
+  return NULL;
 }
 
 /**
- * Create a customer
+ * Create a customer.
  *
- * @param $email
+ * @param string $email
  *   Email of the customer to create.
  *
  * @return array
  *   The result of the request.
  */
-function _ereolen_support_create_customer($email): array {
-  // Always create jira user
-  $requestParts = _ereolen_support_get_request_parts('/customer?strictConflictStatusCode=true', 'POST', ['displayName' => $email, 'email' => $email], 'servicedesk');
-
-  return _ereolen_support_callApi($requestParts);
+function _ereolen_support_create_customer(string $email): array {
+  return _ereolen_support_call_api(
+    '/customer?strictConflictStatusCode=true',
```

Comment:
You should always use https://www.php.net/manual/en/function.http-build-query.php (or something similar, e.g. https://api.drupal.org/api/drupal/includes%21common.inc/function/url/7.x) to build URLs with query strings.